patwey/corkboard
# Refactor repo and project fetching/rendering

## What's this PR do?

**Repo Fetcher**
- Rename to `repo_fetcher` from `project_fetcher`
- Break up nested ajax calls into smaller components

**Repo Renderer**
- Rename to `repo_renderer` from `project_renderer`
- Move repo template into its own file

**Issue Fetcher**
- Break up logic into multiple functions

**Issue Renderer**
- Move issue template into its own file
- Rename functions to be more appropriate
- Consolidate logic of `getIssueType` into `getColumnId`
## Where should the reviewer start?

Start by looking at `repo_fetcher.js` and `repo_renderer.js` - a lot of the changes made here are made in the context of the `issue_fetcher.js` and `issue_renderer.js` but those are more involved.
## How should this be manually tested?
- Fork this repo
- Clone your copy down
- run `$ rails s`
- Check that a repo page (with issues) looks/acts like the production site [corkboard-patwey.herokuapp.com](corkboard-patwey.herokuapp.com) 
## What gif best describes this PR or how it makes you feel?

![](https://media.giphy.com/media/Ydu8M6FcpvPsk/giphy.gif)
